qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1761,7 +1761,16 @@ def filter_fn(node):
         return {tuple(x) for x in group_list}
 
     def collect_1q_runs(self):
-        """Return a set of non-conditional runs of 1q "op" nodes."""
+        """Return a list of runs of 1q gates that are neither classically conditioned nor
+        parameterized.
+
+        Nodes are processed in canonical topological order. Qualifying nodes are 1q gates with a
+        defined numerical matrix that are neither classically conditioned nor parameterized. The
+        first qualifying gate begins the first run. All consecutive qualifying gates are added to
+        this run as long as all gates in the run are on a single wire. Any other node terminates a
+        run, and the next run begins with the next qualifying gate. A qualifying gate terminating a
+        run may be the first gate in the next run.
```

Comment:
I think this is slightly confused, but mostly right.  Getting into the details of the graph traversal is misleading, I think - the description of the output (aside from ordering, which we should treat as arbitrary) can be done without knowing how the algorithm works.

For example: a gate can't both qualify to be in a 1q run and cause a run to terminate.  In the completely separable circuit
```text
q0: -A-B-C-

q1: -D-E-F-
```
it's a valid topological order to visit [A, D, B, E, C, F] and you can write the `collect_runs` algorithm without trouble using any topological ordering.  A run only terminates when there is not exactly one successor, or the single successor doesn't match the filter, regardless of traversal order.  (In practice, I suspect it's most efficient to just follow a wire to a run's completion when you see a run start, but that's an implementation detail.)

I'd consider taking out discussion of the traversal order, and talk about termination of a run only in terms of successor nodes.

Reply:
This brings up a good point. I see "in topological order" in comments and doc strings. But this order is not unique. My understanding is that we have our own canonical topological order. But maybe its only stable in some sense. Furthermore, that point is only potentially important  for describing what's happening. If the nodes were not traversed in topological order (at least for the 2q case) would the consumers of these functions still work properly? I guess so. If so, then we shouldn't document the order, but leave it as an implementation detail. (Never stops people from relying on it, but that's another issue.) ... Maybe a more descriptive rather than procedural characterization of the output would be better.

Reply:
The return value here is (aside from an implementation detail of the order the runs appear in) entirely independent of the node traversal order.  It's an implementation detail that the natural way to write this algorithm is to walk the DAG in an order that happens to be topological.

Reply:
> a gate can't both qualify to be in a 1q run and cause a run to terminate

Right I copy-pasted from the 2q doc string.

Reply:
I assumed that `rx.collect_runs` is the same as `rx.collect_bicolor_runs`, but without the bicolor filter. After actually reading the code. I see it's very different.

Reply:
New commit [`1f5bbd5` (#10246)](https://github.com/Qiskit/qiskit-terra/pull/10246/commits/1f5bbd526b44c8b21b7b7abfd747e425798cdda3)